contrib/local/setup.sh

```diff
@@ -24,47 +26,49 @@ do
 done
 
 # Provide genesis validator self deligations
-$CHAIN_BIN add-genesis-account $(jq -r .genesis[0].address $VALIDATOR_CONFIG) $COINS --keyring-backend="test"
+$CHAIN_BIN add-genesis-account $($CHAIN_BIN keys show -a $(jq -r .genesis[0].name $VALIDATOR_CONFIG) --keyring-backend="test") $COINS --keyring-backend="test"
 
 # Give Validator addresses initial coins
 for ((i=0; i<$(jq -r '.validators | length' $VALIDATOR_CONFIG); i++))
 do
-  $CHAIN_BIN add-genesis-account $(jq -r .validators[$i].address $VALIDATOR_CONFIG) $COINS --keyring-backend="test"
+  $CHAIN_BIN add-genesis-account $($CHAIN_BIN keys show -a $(jq -r .validators[$i].name $VALIDATOR_CONFIG) --keyring-backend="test") $COINS --keyring-backend="test"
 done
 
 for ((i=0; i<$(jq -r '.keys | length' $KEYS_CONFIG); i++))
 do
-  $CHAIN_BIN add-genesis-account $(jq -r .keys[$i].address $KEYS_CONFIG) $COINS --keyring-backend="test"
+  $CHAIN_BIN add-genesis-account $($CHAIN_BIN keys show -a $(jq -r .keys[$i].name $KEYS_CONFIG) --keyring-backend="test") $COINS --keyring-backend="test"
 done
 
-$CHAIN_BIN gentx $(jq -r ".genesis[0].name" $VALIDATOR_CONFIG) 5000000000uxprt --keyring-backend="test" --chain-id $CHAIN_ID
+$CHAIN_BIN gentx $(jq -r ".genesis[0].name" $VALIDATOR_CONFIG) 5000000000$DENOM --keyring-backend="test" --chain-id $CHAIN_ID
+
 echo "Output of gentx"
-cat $HOME/.persistenceCore/config/gentx/*.json | jq
+cat $HOME/$CHAIN_DATA_DIR/config/gentx/*.json | jq
 
 echo "Running collect-gentxs"
 $CHAIN_BIN collect-gentxs
 
+ls $HOME/$CHAIN_DATA_DIR/config
+
 echo "Update app.toml file"
-echo $HOME
-sed -i -e 's#keyring-backend = "os"#keyring-backend = "test"#g' $HOME/.persistenceCore/config/client.toml
+sed -i 's#keyring-backend = "os"#keyring-backend = "test"#g' $HOME/$CHAIN_DATA_DIR/config/client.toml
 
 echo "Update config.toml file"
-sed -i -e 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:26657"#g' $HOME/.persistenceCore/config/config.toml
-sed -i -e 's/timeout_commit = "5s"/timeout_commit = "1s"/g' $HOME/.persistenceCore/config/config.toml
-sed -i -e 's/timeout_propose = "3s"/timeout_propose = "1s"/g' $HOME/.persistenceCore/config/config.toml
-sed -i -e 's/index_all_keys = false/index_all_keys = true/g' $HOME/.persistenceCore/config/config.toml
+sed -i 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:26657"#g' $HOME/$CHAIN_DATA_DIR/config/config.toml
+sed -i 's/timeout_commit = "5s"/timeout_commit = "1s"/g' $HOME/$CHAIN_DATA_DIR/config/config.toml
+sed -i 's/timeout_propose = "3s"/timeout_propose = "1s"/g' $HOME/$CHAIN_DATA_DIR/config/config.toml
+sed -i 's/index_all_keys = false/index_all_keys = true/g' $HOME/$CHAIN_DATA_DIR/config/config.toml
 
 echo "Update genesis.json file with updated local params"
-sed -i -e 's/stake/uxprt/g' $HOME/.persistenceCore/config/genesis.json
+sed -i "s/stake/$DENOM/g" $HOME/$CHAIN_DATA_DIR/config/genesis.json
```

Comment:
@Anmol1696 the -e flag was bug? 
I know the command as is does not work on mac systems.. 
`sed -i "" 's/index_all_keys... ..... ..` -> works

Reply:
ahh my bad, the `-e` flag is the fix not the bug, for the mac systems, let me fix that